app/tests/spec/views/sign_up.js

```diff
@@ -261,10 +261,15 @@ define(function (require, exports, module) {
 
             const $getStartedEl = $suggestSyncEl.find('a');
             assert.equal($getStartedEl.attr('rel'), 'noopener noreferrer');
-            assert.equal($getStartedEl.attr('href'),
+            if (view._webChannelSupportExists()) {
```

Comment:
I have a feeling this unit test isn't doing what you intend it to do. What I *think* you intend for this test to do is to check whether the href is `/signup...` if `_webChannelSupportExists` returns `true` and the other URL if `_webChannelSupportExists` returns `false`. Unfortunately, that's not what's happening, because you are not directly controlling the return value of `_webChannelSupportExists`. The way this is set up, only one of these two branches will ever be taken, the other will always be ignored.

This test needs to be made into three tests.

1. `_webChannelSupportExists` returns `false`. Check that the element href equals `https://mozilla.org/firefox/sync?...`
2. `_webChannelSupportExists` returns `true`, the user is on Firefox for Android. Check that the element href equals `/signup?service=sync&context=fx_fennec_v1&entrypoint=fxa:signup`
3. `_webChannelSupportExists` returns `true`, the user is on Firefox Desktop. Check that the element href equals `/signup?service=sync&context=fx_desktop_v3&entrypoint=fxa:signup`